Refresh.Interfaces.Game/Endpoints/Levels/PublishEndpoints.cs

```diff
@@ -211,6 +213,23 @@ public Response PublishLevel(RequestContext context,
                 return BadRequest;
             }
         }
+        
+        // Automatically mark level as reupload by keyword matching the title
+        // + get original publisher from the description??
+        string[] reuploadWords = { 
+            "(reupload)", "[reupload]",
+            "(archive)", "[archive]"
+        };
+       
+        bool isReUpload = reuploadWords.Any(keyword => level.Title.IndexOf(keyword, StringComparison.OrdinalIgnoreCase) >= 0);
+        if (isReUpload)
+        {
+            level.IsReUpload = true;
+
+            // get publisher -- [OriginalPublisher:{username}]
+            Match attributeMatch = Regex.Match(level.Description, @"\[OriginalPublisher:([^\]]+)\]", RegexOptions.IgnoreCase); // could we turn this in the future into something that allows custom level attributes?
```

Comment:
- This regex needs to be compiled for efficiency, see `Refresh.Common.Verification.CommonPatterns`.
- This pattern is very difficult to type on a controller, try something smaller that doesn't use as many special characters.